src/rtctools/optimization/multi_seed_mixin.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Dict, Iterable
+
+from rtctools.optimization.optimization_problem import OptimizationProblem
+from rtctools.optimization.timeseries import Timeseries
+
+
+class MultiSeedMixin(OptimizationProblem):
+    """
+    Enables a workflow to solve an optimization problem by trying multiple seeds.
+    """
+
+    def __init__(self, **kwargs):
+        self.__selected_seed_id = None
+        super().__init__(**kwargs)
+
+    def use_seed_id(self) -> bool:
+        """Return ``True`` if the selected seed id is used.
+
+        Return ``True`` if the seed corresponding to the selected seed identifier is used
+        for the current run.
+        By default, this is ``True``.
+        Overwrite this method when in some cases the seed is not based on the seed id,
+        e.g. in case of goal programming, return False after the first priority,
+        or in case of homotopy, return ``False`` after the homotopy parameter is increased.
+        """
+        return True
+
+    def selected_seed_id(self) -> Any:
+        """Get the selected seed identifier.
+
+        If the selected seed id is None, use the default seed.
+        """
+        return self.__selected_seed_id
+
+    def seed_ids(self) -> Iterable:
```

Comment:
A simpler way to implement these two methods would be to replace them with a single `seeds` method that return a list of seed dictionaries.

Reply:
Using separate seed ids allows the user to select specific solver options based on the seed id as also mentioned in the `selected_seed_id` discussion. It therefore seems to have added value to keep `seed_ids` and `seed_from_id` separated.

Reply:
See my comments above - I think there is an existing way to achieve the desired solver options.